### ISSUE TITLE
Dropping a table into another table's cell should not inline it

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#9420]
## 📝 Provide detailed reproduction steps (if any)

We're unblocking nesting tables right now and when testing drag and drop I noticed that it may not be intuitive that when you drop a table into another table that dropped table gets inlined.

It may make more sense for pasting and other word processors do that too. However, in case of D&D we're dragging an entire table and it feels like we're dragging an object, so I'd intuitively expect that object to remained untouched.

We'll work on block drag&drop one day (see #7731) and most likely this will be automatically resolved then.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
PS. If you wonder why tables get inlined on paste, there's a separate ticket for that where we explained that this is how e.g. Word works.

**Comment by cbushmak** [comment]
> PS. If you wonder why tables get inlined on paste, there's a separate ticket for that where we explained that this is how e.g. Word works.

Word only inlines it if you have a cell highlighted in the destination table when you paste, otherwise if you just have the cursor in the cell it nests the tables. It would be good if both drag & drop and the Templates plugin nested the tables, even if paste always inlines them, since then you'd at least have options.

**Comment by Witoso** [comment]
Scope:
* change the implementation for drop. When we drop, the table is inserted to the drop target.
* it's integrated into `TableClipboard` which augments the `model.InsertContent`. We don't know if it's paste/drop.
* We have an event in the `ClipboardPipeline`, `contentInsertion`, and it knows if it's paste or drop.

To check:
* how it will work with track changes (most likely it will do the inlining anyway). So we would need to change it in track changes.

**Comment by Witoso** [comment]
Here's the mentioned behavior from Word:

https://github.com/ckeditor/ckeditor5/assets/9881379/51b7da18-70a3-4c31-a962-46f8ade48423

1. Copying and pasting a few cells will inline it in a target table (CKE5 will do the same)
2. Copying all cells will inline it in a target table (CKE5 will do the same)
3. Copying a table, will place it in a cell in a target table (CKE5 **not do the same**, and will inline).

Why no. 3 was not done this way. AFAIK, the problematic part for us is pasting from other sources: Excel, Word, Google Sheet, directly to our CKE5 tables. All the sources mentioned will mark the elements as part of `table` in the clipboard. Example: if you would copy one cell in Google Sheets, and paste it into the editor, it would add a new table inside the cell with one cell.

Not the best UX, and we don't have an easy way of determining if it's a whole table or part of a table. That's why we would need to either inline always or always create a new table inside on paste. (Another option is creating a dedicated UI pasting experience where a user would decide between "Inline" / "Paste table").

The OP describes a change only to the drag and drop action, so that dragging the entire table to a cell, would keep the full table.

This needs another review and discovery, what would be a better behavior.

**Comment by Witoso** [comment]
Drag and dropping tables (and nesting them this way) were fixed in the v45.
